### PR TITLE
New downstream eval

### PR DESCRIPTION
**Pull request by mrudat-iais**
# What does this PR do?
We can now convert a mamba modalities model into a HF model. This model can then be evaluated in eval harness. 
* (**main**) We now have a `convert_pytorch_to_hf_checkpoint` endpoint that takes the `model_name`, `model_config_path` and `output_dir` and creates HF files that can then be loaded with `AutoModel.from_pretrained`
* (**tests**) we have a test that shows the functionality [tests/checkpointing/test_checkpoint_conversion.py](https://github.com/Modalities/modalities/blob/019b8074f79efefa484c580824b4bf9b60ace5da/tests/checkpointing/test_checkpoint_conversion.py)

## General changes
* none besides the above

## Breaking Changes
* no breaking changes

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue / enhancement in isolation
- [x] I have merged main into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have fixed all failing tests (`python tests/tests.py`)

### CODE REVIEW

**Comment by mrudat-iais** [comment]
Hi @fromm-m @le1nux ,

- class added that defines what is needed for the models to be evaluated by eval harness
- all merge conflicts are resolved
- all tests run through

Can one of you please review? Thanks!

File: src/modalities/models/huggingface_adapters/mamba_hf_adapter.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        config.config["model"]["config"]["mixer_model_config"] = MixerModelConfig(
+            **config.config["model"]["config"]["mixer_model_config"])
+        return config.config["model"]["config"]
+
+    def convert_config_config_to_pydantic(self, config):
+        config["is_encoder_decoder"] = False
+        return MambaHuggingFaceModelAdapterConfig(**config)
+
+
+class MambaHuggingFaceModelAdapterConfig(HuggingFaceModelAdapterConfig):
```
Comment (le1nux) [inline src/modalities/models/huggingface_adapters/mamba_hf_adapter.py:48]: I looks like we could simplify this via e.g., 

```python
class MambaHuggingFaceModelAdapterConfig(HuggingFaceModelAdapterConfig, MambaLLMConfig):
    is_encoder_decoder: Literal[False] = False

```

**Review by le1nux — CHANGES_REQUESTED** [review]

**Review by le1nux — COMMENTED** [review]
Added a bunch of remarks mostly concerning code standards. I did not an exhaustive list of issues, so please also look into the remaining source files  for similar issues such as unused imports, missing type hints, lack of comments. Thanks! :-)

File: src/modalities/checkpointing/checkpoint_conversion.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def __init__(
+            self, config_file_path: Path, output_hf_checkpoint_dir: Path
+    ):
+        self.output_hf_checkpoint_dir = output_hf_checkpoint_dir
+
+        if not config_file_path.exists():
+            raise ValueError(f"Could not find {config_file_path}")
+
+        self.config_dict = load_app_config_dict(config_file_path)
+        logging.info(f"Config\n{self.config_dict}")
```
[the remark this answers is not in the document]
Comment (mrudat-iais) [reply src/modalities/checkpointing/checkpoint_conversion.py:23]: why should we remove it?

**Pull request closed without merging**